File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/plugins/angularjs/service/goodsService.js
app.service('goodsService', function ($http) {

    this.add = function (entity) {
        return $http.post('../goods/add.do', entity);
    };

    this.update = function (entity) {
        return $http.post('../goods/update.do', entity);
    };

    this.findByPage = function (page, size, searchEntity) {
        return $http.post('../goods/findByPage.do?page=' + page + '&size=' + size, searchEntity);
    };

    this.findById = function (id) {
        return $http.post('../goods/findByGoodsId.do?goodsId=' + id);
    };

    this.updateStatus = function (ids, status) {
        return $http.post('../goods/updateStatus/' + status + '.do?ids=' + ids)
    };

    this.del = function (ids) {
        return $http.get('../goods/del.do?ids=' + ids);
    };
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/GoodsService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbGoods;
import com.pinyougou.pojo.TbItem;
import com.pinyougou.vo.GoodsVo;
import entity.PageResult;

import java.util.List;

public interface GoodsService {

    void add(GoodsVo goodsVo) throws Exception;

    PageResult<TbGoods> findByPage(TbGoods goods, int pageNum, int pageSize) throws Exception;

    GoodsVo findByGoodsId(Long id) throws Exception;

    TbGoods findById(Long id) throws Exception;

    void update(GoodsVo goodsVo) throws Exception;

    void updateStatus(Long[] ids, String status) throws Exception;

    void del(Long[] ids) throws Exception;

    void updateMarketable(Long[] ids, String isMarketable) throws Exception;

    /**
     * 根据商品ID和状态查询Item表信息
     *
     * @param goodsIds
     * @param status
     * @return
     */
    public List<TbItem> findItemListByGoodsIdandStatus(Long[] goodsIds, String status) throws Exception;
}